### PR TITLE
9. Create a "perfect" algorithm to predict the weather. // Elias, Anton

### PR DESCRIPTION
**Pull request by antonsvaren**
Created a function to calculate the span of temperatures with a confidence interval of 90%.

//Elias, Anton

### CODE REVIEW

File: uWeather/app/src/main/java/dit257/mandalore/uweather/Probability.kt [hunk 1]
```diff
@@ -1,36 +1,36 @@
 package dit257.mandalore.uweather
 
+import java.lang.Math.*
 import java.util.*
+import kotlin.math.abs
 
 class Probability {
 
     //Calculate probability based on difference from median
```
Comment (gu92maak) [inline uWeather/app/src/main/java/dit257/mandalore/uweather/Probability.kt]: please make a more clear description that it takes an array of temperatures from different weather services to calculate a probable (is aggregated the right word?) forecast.

File: uWeather/app/src/main/java/dit257/mandalore/uweather/Probability.kt [hunk 2]
```diff
@@ -1,36 +1,36 @@
...
-        for (i in provider){
-            //Get min/max temperature reported
-            if(i<min){min=i}
-            if(actual_temp.get(nrOfReadings-1)<min){min=i}
-            if(i>max){max=i}
-            if(actual_temp.get(nrOfReadings-1)>max){max=i}
-
-            nrOfReadings++
-            difference += i - actual_temp.get(nrOfReadings-1)
+    // Calculate the span of temperatures, with a confidence interval of 90%.
```
Comment (gu92maak) [inline uWeather/app/src/main/java/dit257/mandalore/uweather/Probability.kt]: please make a more clear description that it takes an array of temperatures from different weather services to calculate a probable (is aggregated the right word?)  confidence intervall.

File: uWeather/app/src/main/java/dit257/mandalore/uweather/Probability.kt [hunk 3]
```diff
@@ -1,36 +1,36 @@
 package dit257.mandalore.uweather
 
+import java.lang.Math.*
 import java.util.*
+import kotlin.math.abs
 
 class Probability {
 
     //Calculate probability based on difference from median
-    fun calcProbabilityMedian(all_last_temps:IntArray): Double {
```
Comment (gu92maak) [inline uWeather/app/src/main/java/dit257/mandalore/uweather/Probability.kt:8]: does this only work for the last Temperatures and not for future temperatures?

File: uWeather/app/src/main/java/dit257/mandalore/uweather/Probability.kt [hunk 4]
```diff
@@ -1,36 +1,36 @@
...
-        var total = 0
-        for(t in all_last_temps){
+    fun calcProbabilityMean(allLastTemps:DoubleArray, temp:Double): Double {
```
Comment (gu92maak) [inline uWeather/app/src/main/java/dit257/mandalore/uweather/Probability.kt:10]: does this only work for the last Temperatures and not for future temperatures?

File: uWeather/app/src/main/java/dit257/mandalore/uweather/Probability.kt (hunk 3 above)
Comment (antonsvaren) [reply uWeather/app/src/main/java/dit257/mandalore/uweather/Probability.kt:8]: This only works for the current temperature, a.k.a the latest temperature provided by the API request

Comment (antonsvaren) [reply uWeather/app/src/main/java/dit257/mandalore/uweather/Probability.kt:8]: Resolve the conversations when you have understood :)

File: uWeather/app/src/main/java/dit257/mandalore/uweather/Probability.kt (hunk 2 above)
Comment (antonsvaren) [reply uWeather/app/src/main/java/dit257/mandalore/uweather/Probability.kt]: Okay, I can update the comments and see if it is more understandable

File: uWeather/app/src/main/java/dit257/mandalore/uweather/Probability.kt (hunk 1 above)
Comment (antonsvaren) [reply uWeather/app/src/main/java/dit257/mandalore/uweather/Probability.kt]: Will update the comments :)

File: uWeather/app/src/main/java/dit257/mandalore/uweather/Probability.kt (hunk 4 above)
Comment (antonsvaren) [reply uWeather/app/src/main/java/dit257/mandalore/uweather/Probability.kt:10]: This only works for the current temperature, a.k.a the latest temperature provided by the API request

**Review by MaidScientistIzutsumiMarin — COMMENTED** [review]
The math hurts my head. It seems good.

File: uWeather/app/src/main/java/dit257/mandalore/uweather/Probability.kt (hunk 4 above)
Comment (gu92maak) [reply uWeather/app/src/main/java/dit257/mandalore/uweather/Probability.kt:10]: I understand and agree

File: uWeather/app/src/main/java/dit257/mandalore/uweather/Probability.kt (hunk 3 above)
Comment (gu92maak) [reply uWeather/app/src/main/java/dit257/mandalore/uweather/Probability.kt:8]: I understand and agree

**Pull request merged**